### DISCUSSION TITLE
Modal with Browser wallets + ethers?

### DISCUSSION THREAD

**Discussion by PaperMold** [WalletConnect/walletconnect-monorepo#3823] [Developer Support] [answered]
Hey there,
First of all thank you for all the work you guys been putting on this project!

I was using WalletConnect v1 before and I'm still trying to understand how can I migrate to a more updated version and have the same functionality, use your modal with ethers and have the max compatibility with other wallets, mobile and browser.

I've noticed that's possible with '@walletconnect/ethereum-provider', however it looks like it isn't compatible with metamask in browser, even though Coinbase in browser works. Is it possible to edit the modal to add a custom button so I can integrate metamask (browser) in the modal?

I've also tried Web3Modal SDK with React but it works with hooks instead of raw functions, if I want to make a transaction to buy X hooks aren't the thing I would like to use.

What should I go for in this case?
Thanks in advance!

**Comment by glitch-txs** [comment]
Hooks are important if you're using react because the provider will change depending on the users connection state. However we could expose a getter for the provider/signer. Would that be helpful? Could you show a code snippet on how you would like it to work?

**Comment by PaperMold** [comment]
Hey @glitch-txs,

Exposing the provider/signer would solve my issue, yes!

My project is a decentralized dApp and therefore we make a lot of contract calls, both view and transactional.
To prevent unnecessary requests we throttle and cache a lot of these calls. We use react contexts to manage this cache/throttle system.

By exposing the provider/signer I'd only need to do small changes to keep the exact same functionality, I'd only need to use the signer from web3modal.

**Reply by glitch-txs** [reply, marked as answer]
can you try this:
```ts
export const modal = createWeb3Modal({ wagmiConfig, projectId, chains })
```
then use these
https://docs.walletconnect.com/web3modal/javascript/actions?platform=ethers#subscribe-to-theme-changes

**Reply by PaperMold** [reply]
Web3Modal SDK + Ethers works for my case 👍 
The only thing I would change is update to ethers v6 if I could but this definitely works.

Thanks for your help :)

**Reply by glitch-txs** [reply]
we are currently working on v6 integration, thanks for the feedback! 🤝
